### PR TITLE
Multiple changes to the validity package.

### PR DESCRIPTION
**Pull request by yuizumi**
- Add FixedURL: use the same validity URL for all resources.
- Rename ValidityURLRule to URLRule to reduce the redundancy; it makes golint happier.
- Split url.go into url_rule.go and append_ext.go; having all rules in a single source file makes it too large.
- Use http.ParseTime instead of httpheader.ParseDate (our own code).

### CODE REVIEW

File: validity/url_rule.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+// URLRule decides the validity URL of a resource.
+type URLRule interface {
+	// Apply returns the validity URL of a resource. physurl is the physical
+	// URL of the resource; resp is the HTTP response; vp is the period the
+	// signed exchange will be valid for. The physical URL is typically equal
+	// to the request URL but different in some cases; see package urlrewrite
+	// for more details.
+	//
+	// Note ValidityURLRule implementations can retrieve the request URL via
```
Comment (banaag) [inline validity/url_rule.go:31]: ValidityURLRule -> URLRule?

Comment (yuizumi) [reply validity/url_rule.go]: Oops, good catch.

**Comment by yuizumi** [comment]
The build is now fixed (by #42).

**Comment by yuizumi** [comment]
Thanks!

**Pull request merged**